Sources/X509/Extension Types/ExtendedKeyUsage.swift

```diff
@@ -26,8 +26,20 @@ public struct ExtendedKeyUsage {
     ///
     /// - Parameter usages: The purposes for which the certificate may be used.
     @inlinable
-    public init<Usages: Sequence>(_ usages: Usages) where Usages.Element == Usage {
+    public init<Usages: Sequence>(_ usages: Usages) throws where Usages.Element == Usage {
         self.usages = Array(usages)
+        
+        // check for duplicates using a linear scan
+        // this is more performant compared to hashing if we have less than ~64 usages
+        for index in self.usages.indices {
+            let usage = self.usages[index]
+            for currentIndex in self.usages.indices.dropFirst(index + 1) {
+                let currentUsage = self.usages[currentIndex]
+                if usage == currentUsage {
+                    throw CertificateError.duplicateOID(reason: "duplicate \(usage) usage. First at \(index) and second at \(currentIndex)")
+                }
+            }
+        }
```

Comment:
We've written this code twice: can you factor the algorithm out to a common helper function and then call it in the relevant places?

Reply:
Done in https://github.com/apple/swift-certificates/pull/90/commits/cdf557e0cd72defa7b62cb6798dfe1ad3334e0fc